### PR TITLE
Add support for POINTING_DEVICE_DRIVER

### PR DESCRIPTION
**Pull request by drashna**
## Description

Like the title says... and some cleanup. 

## Types of Changes

- [x] Core
- [x] New feature
- [x] Enhancement/optimization
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Creating as a draft for now, since I'd like to add the pmw3389 sensor, as well.

File: drivers/optical/pmw3360_firmware.h [hunk 1]
```diff
@@ -18,6 +18,9 @@
 
 #pragma once
 
+#define PMW3360_PRODUCT_ID         0x42
```
Comment (tzarc) [inline drivers/optical/pmw3360_firmware.h:21]: ....where did this firmware originate? A firmware blob without a source seems...questionable.

Comment (drashna) [reply drivers/optical/pmw3360_firmware.h:21]: Well, fun times.  The firmware blob is from here: https://github.com/mrjohnk/PMW3360DM-T2QU

Which appears to be the repo for the tindie shop seller for the sensor.  So "without a source" seems to be the real answer.  Though, at least, should include said links in the file.

File: drivers/avr/spi_master.h [hunk 2]
```diff
@@ -39,19 +39,15 @@ typedef int16_t spi_status_t;
 #ifdef __cplusplus
 extern "C" {
 #endif
-void spi_init(void);
-
-bool spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint16_t divisor);
```
Comment (fauxpark) [inline drivers/avr/spi_master.h:42]: ?

Comment (drashna) [reply drivers/avr/spi_master.h:42]: honestly, not sure. Merge issue probably.   Also, definitely in progress

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

**Comment by drashna** [comment]
Depends on #13044

**Pull request closed without merging**